soroban-env-host/src/fees.rs

```diff
@@ -53,11 +53,53 @@ pub struct FeeConfiguration {
     pub fee_per_propagate_1kb: i64,
 }
 
+/// Change in a single ledger entry with parameters relevant for rent fee
+/// computations.
+///
+/// This represents the entry state before and after transaction has been
+/// applied.
+pub struct LedgerEntryRentChange {
+    /// Whether this is persistent or temporary entry.
+    pub is_persistent: bool,
+    /// Size of the entry in bytes before it has been modified.
+    /// `0` for newly-created entires.
+    pub old_size_bytes: u32,
+    /// Size of the entry in bytes after it has been modified.
+    pub new_size_bytes: u32,
+    /// Expiration ledger of the entry before it has been modified.
+    /// `0` for newly-created entires.
```

Comment:
Should this be current ledger - 1 instead of 0 so you don't need to handle the 0 case? Or is that how you distinguish a new entry from an old one?

Reply:
Anything less than current ledger would work actually, there is no special logic based on this. `0` is just for consistency with 0 for `old_size_bytes`.

Reply:
Updated comment.